internal/cluster/discovery_it_test.go

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2008-2023, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cluster_test
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/hazelcast/hazelcast-go-client/cluster/discovery"
+	"github.com/hazelcast/hazelcast-go-client/internal/it"
+)
+
+func TestDiscoveryStrategy(t *testing.T) {
+	st := &staticDiscoveryStrategy{}
+	tcx := it.MapTestContext{
+		T: t,
+		ConfigCallback: func(tcx it.MapTestContext) {
+			st.addrs = tcx.Cluster.DefaultConfig().Cluster.Network.Addresses
```

Comment:
Can we set Network.Addresses to nil to ensure discovery works?

Reply:
Sure, at: https://github.com/hazelcast/hazelcast-go-client/pull/937/commits/1f2ad596fc07b107142873a9280f22f94071f0f5